### DISCUSSION TITLE
[Question][Hw1-1] Q1. About exception raising

### DISCUSSION THREAD

**Discussion by sjoon2455** [prosyslab-classroom/cs524-program-analysis#410] [Q&A]
Hi, do I have to implement something like try...except... for this problem?

I am not quite sure what does it mean by "The function will raise ... when the arguments are invalid".

Could you please clarify this?

Thanks in advance.

**Comment by hyunsukimsokcho** [comment]
By saying `try .. except ..`, I assume you are referring to exception handling in python. However, in the homework, one does not have to handle any exception, but simply throwing the exception is enough. If you are familiar with python, `raise` is the one throwing the exception. One may search about how exception is defined and how can it be thrown in OCaml.

**Comment by sjoon2455** [comment]
@hyunsukimsokcho, thanks for answering.

I guess ```raise exception Invalid_argument``` is what I need at this point. 
However, if I run the implemented function with invalid arguments, for instance ```sigma "1" 10``` or ```sigma 1.1 10```, it does not throw Invalid_argument error. Could you guide me a bit more?

Thanks.

**Comment by hyunsukimsokcho** [comment]
(I assume you are referring to problem 1) Those two cases you mentioned will raise type error even before executing your code. Since OCaml is strongly-typed language, two arguments for `sigma` should be `int`. However, even if one passes correctly typed arguments, there are cases where usual `sigma` cannot be computed properly. For those cases, throw exception.

**Comment by hyunsukimsokcho** [comment]
minor correction: `sigma "1" 10` and `sigma 1.1 10` will execute your code however, they will throw type error when trying to evaluate `+` in proper implementation of `sigma`. Note that `+` in OCaml only works with two arguments of which type is `int`.

**Comment by sjoon2455** [comment]
@hyunsukimsokcho, I understood your point! Thank you. Issue resolved, so closing this issue.
